modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Hash.kt

```diff
@@ -0,0 +1,45 @@
+package arrow.typeclasses
+
+/**
+ * A type class used to represent hashing for objects of type [F]
```

Comment:
We should probably explain what a Hash is here

Reply:
What a hash value is and what they are useful for. A simple introductory paragraph should be enough IMO.

Reply:
Added something. Not sure if enough, or good. :see_no_evil: